Program fails when incorrect GitHub token is used

## Describe the bug

The program fails when an incorrect GitHub token is used.

invalid token it is not caught in `ActionInputs().validate_repositories_configuration()` and it failes on [this](https://github.com/AbsaOSS/living-doc-generator/blob/6f197666eff77b06a687951d6e6cac2128b51aec/living_documentation_regime/living_documentation_generator.py#L75) line.

```
self = <github.Auth.Token object at 0x1057d9820>, token = ''

    def __init__(self, token: str):
        assert isinstance(token, str)
>       assert len(token) > 0
E       AssertionError
```

## To Reproduce

Set GH token to an empty string

```
    os.environ["INPUT_GITHUB_TOKEN"] = ""
```

--- a/living_documentation_regime/action_inputs.py
+++ b/living_documentation_regime/action_inputs.py
@@ -136,9 +136,13 @@ class ActionInputs:
         # Validate GitHub token
         response = requests.get("https://api.github.com/octocat", headers=headers, timeout=10)
         if response.status_code == 401:
-            logger.error("Invalid GitHub token. Please verify that the token is correct.")
+            logger.error("Can not connect to GitHub. Possible cause: Invalid GitHub token. Please verify that the token is correct.",
+                         response.status_code,
+                         response.text,
+                         )
             sys.exit(1)
 
+        repository_error_count = 0
         for repository in repositories:
             org_name = repository.organization_name
             repo_name = repository.repository_name
@@ -153,7 +157,7 @@ class ActionInputs:
                     org_name,
                     repo_name,
                 )
-                sys.exit(1)
+                repository_error_count += 1
             elif response.status_code != 200:
                 logger.error(
                     "An error occurred while validating the repository '%s/%s'. The response status code is %s.",
@@ -162,6 +166,8 @@ class ActionInputs:
                     response.status_code,
                     response.text,
                 )
-                sys.exit(1)
+                repository_error_count += 1
+        if repository_error_count > 0:
+            sys.exit(1)
 
         logger.debug("Repositories configuration validation successfully completed.")

--- a/living_documentation_regime/action_inputs.py
+++ b/living_documentation_regime/action_inputs.py
@@ -133,6 +133,12 @@ class ActionInputs:
         github_token = self.get_github_token()
         headers = {"Authorization": f"token {github_token}"}
 
+        # Validate GitHub token
+        response = requests.get("https://api.github.com/octocat", headers=headers, timeout=10)
+        if response.status_code == 401:
+            logger.error("Invalid GitHub token. Please verify that the token is correct.")
+            sys.exit(1)
+
         for repository in repositories:
             org_name = repository.organization_name
             repo_name = repository.repository_name
@@ -146,6 +152,15 @@ class ActionInputs:
                     "exists and that your authorization token is correct.",
                     org_name,
                     repo_name,
+                )
+                sys.exit(1)
+            if response.status_code != 200:
+                logger.error(
+                    "An error occurred while validating the repository '%s/%s'. The response status code is %s.",
+                    org_name,
+                    repo_name,
+                    response.status_code,
+                    response.text
                 )
                 sys.exit(1)
 

--- a/living_documentation_regime/action_inputs.py
+++ b/living_documentation_regime/action_inputs.py
@@ -135,11 +135,12 @@ class ActionInputs:
 
         # Validate GitHub token
         response = requests.get("https://api.github.com/octocat", headers=headers, timeout=10)
-        if response.status_code == 401:
-            logger.error("Can not connect to GitHub. Possible cause: Invalid GitHub token. Please verify that the token is correct.",
-                         response.status_code,
-                         response.text,
-                         )
+        if response.status_code != 200:
+            logger.error(
+                "Can not connect to GitHub. Possible cause: Invalid GitHub token. Please verify that the token is correct.",
+                response.status_code,
+                response.text,
+            )
             sys.exit(1)
 
         repository_error_count = 0

--- a/living_documentation_regime/action_inputs.py
+++ b/living_documentation_regime/action_inputs.py
@@ -160,7 +160,7 @@ class ActionInputs:
                     org_name,
                     repo_name,
                     response.status_code,
-                    response.text
+                    response.text,
                 )
                 sys.exit(1)
 

--- a/living_documentation_regime/action_inputs.py
+++ b/living_documentation_regime/action_inputs.py
@@ -154,7 +154,7 @@ class ActionInputs:
                     repo_name,
                 )
                 sys.exit(1)
-            if response.status_code != 200:
+            elif response.status_code != 200:
                 logger.error(
                     "An error occurred while validating the repository '%s/%s'. The response status code is %s.",
                     org_name,

--- a/tests/unit/living_documentation_regime/test_action_inputs.py
+++ b/tests/unit/living_documentation_regime/test_action_inputs.py
@@ -238,9 +238,9 @@ def test_validate_repositories_configuration_wrong_configuration(mocker, config_
     )
     mocker.patch("living_documentation_regime.action_inputs.ActionInputs.get_github_token", return_value="fake-token")
     mock_exit = mocker.patch("sys.exit")
-    fake_correct_response = mocker.Mock()
-    fake_correct_response.status_code = 404
-    mocker.patch("living_documentation_regime.action_inputs.requests.get", return_value=fake_correct_response)
+    mock_error_response = mocker.Mock()
+    mock_error_response.status_code = 404
+    mocker.patch("living_documentation_regime.action_inputs.requests.get", return_value=mock_error_response)
 
     # Act
     ActionInputs().validate_repositories_configuration()
@@ -260,11 +260,17 @@ def test_validate_repositories_wrong_token(mocker, config_repository):
     mocker.patch("living_documentation_regime.action_inputs.ActionInputs.get_github_token", return_value="")
     mock_exit = mocker.patch("sys.exit")
 
+    mocker.patch(
+        "living_documentation_regime.action_inputs.ActionInputs.get_repositories", return_value=list()
+    )
+
     # Act
     ActionInputs().validate_repositories_configuration()
 
     # Assert
     mock_exit.assert_called_once_with(1)
     mock_log_error.assert_called_once_with(
-        "Invalid GitHub token. Please verify that the token is correct."
+        "Can not connect to GitHub. Possible cause: Invalid GitHub token. Please verify that the token is correct.",
+         401,
+        '{"message":"Bad credentials","documentation_url":"https://docs.github.com/rest","status":"401"}'
     )

--- a/tests/unit/living_documentation_regime/test_action_inputs.py
+++ b/tests/unit/living_documentation_regime/test_action_inputs.py
@@ -246,8 +246,8 @@ def test_validate_repositories_configuration_wrong_configuration(mocker, config_
     ActionInputs().validate_repositories_configuration()
 
     # Assert
-    mock_exit.assert_called_once_with(1)
-    mock_log_error.assert_called_once_with(
+    assert mock_exit.call_count == 2
+    mock_log_error.assert_any_call(
         "Repository '%s/%s' could not be found on GitHub. Please verify that the repository exists and that your authorization token is correct.",
         "test_org",
         "test_repo",

--- a/tests/unit/living_documentation_regime/test_action_inputs.py
+++ b/tests/unit/living_documentation_regime/test_action_inputs.py
@@ -252,3 +252,19 @@ def test_validate_repositories_configuration_wrong_configuration(mocker, config_
         "test_org",
         "test_repo",
     )
+
+def test_validate_repositories_wrong_token(mocker, config_repository):
+    # Arrange
+    mock_log_error = mocker.patch("living_documentation_regime.action_inputs.logger.error")
+
+    mocker.patch("living_documentation_regime.action_inputs.ActionInputs.get_github_token", return_value="")
+    mock_exit = mocker.patch("sys.exit")
+
+    # Act
+    ActionInputs().validate_repositories_configuration()
+
+    # Assert
+    mock_exit.assert_called_once_with(1)
+    mock_log_error.assert_called_once_with(
+        "Invalid GitHub token. Please verify that the token is correct."
+    )

--- a/tests/unit/test_main.py
+++ b/tests/unit/test_main.py
@@ -24,6 +24,10 @@ from utils.constants import OUTPUT_PATH
 
 def test_run_correct_behaviour_with_all_regimes_enabled(mocker):
     # Arrange
+    fake_correct_response = mocker.Mock()
+    fake_correct_response.status_code = 404
+    mocker.patch("living_documentation_regime.action_inputs.requests.get", return_value=fake_correct_response)
+
     expected_output_path = os.path.abspath(OUTPUT_PATH)
     mock_log_info = mocker.patch("logging.getLogger").return_value.info
     mock_living_doc_generator = mocker.patch("main.LivingDocumentationGenerator")
@@ -55,6 +59,10 @@ def test_run_correct_behaviour_with_all_regimes_enabled(mocker):
 
 def test_run_with_zero_regimes_enabled(mocker):
     # Arrange
+    fake_correct_response = mocker.Mock()
+    fake_correct_response.status_code = 404
+    mocker.patch("living_documentation_regime.action_inputs.requests.get", return_value=fake_correct_response)
+
     mock_log_info = mocker.patch("logging.getLogger").return_value.info
     mocker.patch.dict(os.environ, {"INPUT_GITHUB_TOKEN": "fake_token", "INPUT_LIV_DOC_REGIME": "false"})
     mock_living_doc_generator = mocker.patch("main.LivingDocumentationGenerator")

--- a/tests/unit/test_main.py
+++ b/tests/unit/test_main.py
@@ -24,9 +24,7 @@ from utils.constants import OUTPUT_PATH
 
 def test_run_correct_behaviour_with_all_regimes_enabled(mocker):
     # Arrange
-    mock_error_response = mocker.Mock()
-    mock_error_response.status_code = 404
-    mocker.patch("living_documentation_regime.action_inputs.requests.get", return_value=mock_error_response)
+    mocker.patch("living_documentation_regime.action_inputs.ActionInputs.validate_repositories_configuration")
 
     expected_output_path = os.path.abspath(OUTPUT_PATH)
     mock_log_info = mocker.patch("logging.getLogger").return_value.info
@@ -59,9 +57,7 @@ def test_run_correct_behaviour_with_all_regimes_enabled(mocker):
 
 def test_run_with_zero_regimes_enabled(mocker):
     # Arrange
-    mock_error_response = mocker.Mock()
-    mock_error_response.status_code = 404
-    mocker.patch("living_documentation_regime.action_inputs.requests.get", return_value=mock_error_response)
+    mocker.patch("living_documentation_regime.action_inputs.ActionInputs.validate_repositories_configuration")
 
     mock_log_info = mocker.patch("logging.getLogger").return_value.info
     mocker.patch.dict(os.environ, {"INPUT_GITHUB_TOKEN": "fake_token", "INPUT_LIV_DOC_REGIME": "false"})

--- a/tests/unit/test_main.py
+++ b/tests/unit/test_main.py
@@ -24,9 +24,9 @@ from utils.constants import OUTPUT_PATH
 
 def test_run_correct_behaviour_with_all_regimes_enabled(mocker):
     # Arrange
-    fake_correct_response = mocker.Mock()
-    fake_correct_response.status_code = 404
-    mocker.patch("living_documentation_regime.action_inputs.requests.get", return_value=fake_correct_response)
+    mock_error_response = mocker.Mock()
+    mock_error_response.status_code = 404
+    mocker.patch("living_documentation_regime.action_inputs.requests.get", return_value=mock_error_response)
 
     expected_output_path = os.path.abspath(OUTPUT_PATH)
     mock_log_info = mocker.patch("logging.getLogger").return_value.info
@@ -59,9 +59,9 @@ def test_run_correct_behaviour_with_all_regimes_enabled(mocker):
 
 def test_run_with_zero_regimes_enabled(mocker):
     # Arrange
-    fake_correct_response = mocker.Mock()
-    fake_correct_response.status_code = 404
-    mocker.patch("living_documentation_regime.action_inputs.requests.get", return_value=fake_correct_response)
+    mock_error_response = mocker.Mock()
+    mock_error_response.status_code = 404
+    mocker.patch("living_documentation_regime.action_inputs.requests.get", return_value=mock_error_response)
 
     mock_log_info = mocker.patch("logging.getLogger").return_value.info
     mocker.patch.dict(os.environ, {"INPUT_GITHUB_TOKEN": "fake_token", "INPUT_LIV_DOC_REGIME": "false"})

